### PR TITLE
Couch to SQL forms & cases fixes and cleanup

### PR DESCRIPTION
**Pull request by millerdev**
Review by commit 🐟 

@proteusvacuum @snopoke

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

**Comment by snopoke** [comment]
Build failing: `corehq.apps.couch_sql_migration.tests.test_migration:MigrationTestCase.test_migration_blacklist`

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 1]
```diff
@@ -426,12 +437,19 @@ def _diff_ledgers(self, case_ids):
...
 
-    def _assert_no_migration_restrictions(self, domain_name):
-        assert should_use_sql_backend(domain_name)
-        assert not COUCH_SQL_MIGRATION_BLACKLIST.enabled(domain_name, NAMESPACE_DOMAIN)
-        assert not any(custom_report_domain == domain_name
-                       for custom_report_domain in settings.DOMAIN_MODULE_MAP.keys())
-        assert not REMINDERS_MIGRATION_IN_PROGRESS.enabled(domain_name)
+    def _check_for_migration_restrictions(self, domain_name):
+        if not should_use_sql_backend(domain_name):
+            msg = "does not have SQL backend not enabled"
```
Comment (proteusvacuum) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: nit this is a double negative, but I think you meant `does not have SQL backend enabled`

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 1 above)
Comment (millerdev) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: https://github.com/dimagi/commcare-hq/pull/24061/commits/850c7777d133e86a5b1d70f9badb2181d9e4d65a

**Comment by millerdev** [comment]
✅

**Review by mkangia — APPROVED** [review]
reviewed last 4 commits, looks good

**Pull request merged**